lib/route-recognizer.js

```diff
@@ -249,7 +244,7 @@ State.prototype = {
 // Sort the routes by specificity
 function sortSolutions(states) {
   return states.sort(function(a, b) {
-    return b.specificity.val - a.specificity.val;
+    return (b.specificity.val < a.specificity.val) ? -1 : (b.specificity.val === a.specificity.val) ? 0 : 1;
```

Comment:
This modification leaves it as a string comparison which gives us the behavior that we want.
[See section 11.8.5.4 of the ECMAScript specification.](http://www.ecma-international.org/ecma-262/5.1/#sec-11.8.5)